planning/how-why/src/zenhubTools.js

```diff
@@ -0,0 +1,238 @@
+export const issueUrl = ({ repository: { ownerName, name }, number }) =>
+  `https://github.com/${ownerName}/${name}/issues/${number}`;
+
+/**
+ * @typedef {{
+ *   releases: { nodes: {
+ *     id: string,
+ *     endOn: string,
+ *     title: string,
+ *     issues: { nodes: IssueKey[] }
+ *   }[]}
+ *   repositoriesConnection: { nodes: {
+ *     ghId: number,
+ *     ownerName: string,
+ *     name: string,
+ *   }[]}
+ * }} WorkspaceInfo
+ *
+ * @typedef {{
+ *   repository: {
+ *     ownerName: string,
+ *     name: string,
+ *     ghId: number,
+ *   },
+ *   number: number,
+ * }} IssueKey
+ * @typedef {IssueKey & {title: string, state: 'OPEN' | 'CLOSED'}} IssueLeaf
+ *
+ * @typedef {IssueLeaf & {
+ *  blockedIssues: { nodes: IssueLeaf[] },
+ *  blockingIssues: { nodes: IssueLeaf[] },
+ *  epic: { childIssues: { nodes: IssueLeaf[] }},
+ *  body: string,
+ *  labels: { nodes: { name: string, color: string}[] },
+ * }} IssueInfo
+ */
+
+/**
+ * authoring tool: https://developers.zenhub.com/explorer
+ */
+const queries = {
+  workspace: `
+query workspaceReleases($id: ID!) {
+  workspace(id: $id) {
+    name
+    defaultRepository {
+      ghId
+      ownerName
+      name
+    }
+    releases(state: {eq: OPEN}) {
+      nodes {
+        id
+        state
+        endOn
+        title
+        issues {
+          totalCount
+          nodes {
+            repository { ghId ownerName name }
+            number
+          }
+        }
+      }
+    }
+    repositoriesConnection {
+      nodes {
+        ownerName
+        name
+        ghId
+      }
+    }
+  }
+}`,
+  issue: `
+query getIssueInfo($repositoryGhId: Int!, $issueNumber: Int!) {
+  issueByInfo(repositoryGhId: $repositoryGhId, issueNumber: $issueNumber) {
+    state
+    number
+    title
+    repository {
+      ghId
+      ownerName
+      name
+    }
+    epic {
+      childIssues {
+        nodes {
+          number
+          state
+          title
+          repository {
+            ghId
+            ownerName
+            name
+          }
+        }
+      }
+    }
+    blockedIssues {
+      nodes {
+        number
+        state
+        title
+        repository {
+          ghId
+          ownerName
+          name
+        }
+      }
+    }
+    blockingIssues {
+      nodes {
+        number
+        state
+        title
+        repository {
+          ghId
+          ownerName
+          name
+        }
+      }
+    }
+    body
+    labels(first: 10) {
+      nodes {
+        name
+        color
+      }
+    }
+  }
+}
+`,
+};
+
+/**
+ * @param {string} id
+ * @param {ReturnType<typeof import('./graphql').endpoint>} endpoint
+ */
+export const queryWorkspace = async (id, endpoint) => {
+  const result = await endpoint.query(queries.workspace, { id });
+  if (!("data" in result)) {
+    throw Error("queryWorkspace: no data");
+  }
+  return result.data.workspace;
+};
+
+/** @param {Map<string, IssueInfo>} issueDetail */
+export const issueDepDot = (issueDetail) => {
```

Comment:
the other exports of this package are for using the Zenhub API. consider `viz.js` or something like that for this one